docs/quickstart-upgrade.md

```diff
@@ -0,0 +1,106 @@
+# Setup
+
+## CORS & CSP
+
+The GetSetUp Embedded product requires some CORS and CSP configuration by GetSetUp to allow other sites access. We have configured access for http://localhost:3001. If you require access for other development urls please contact deepak@getsetup.io.
+
+## Site layout
+
+For the example below we will assume a two page site. The first page will host the "discovery" page that embeds a list of classes offered by GetSetUp.
+The second page is the "watch" page that hosts the video viewing and chat experience.
+
+## How to install locally
+
+In your project, run `npm install @getsetup/embed`
+The next step is to import and configure the package on a couple of pages.
+
+### Discovery page
+
+First on a page where you want the class browsing experience create a div and setup the iframe:
+
+```html
+<div id="iframe-target"></div>
+```
+
+```ts
+import * as GSU from '@getsetup/embed'
+
+// This function is supplied by the hosting page.
+// This function will be called by the GSU iframe when it needs to navigate.
+const navigationCallBack = (navigationAction: GSU.NavigationAction, classId: string) => {
+  let href = `/hostSitePath/${navigationAction}`
+
+  // The 'watch' navigation action is special because we need to pass the classId to the 'watch' page.
+  if (navigationAction === 'watch') {
+    href = href + `?classId=${classId}`
+  }
+  window.location.href = href
+}
+
+// Token exchange is not yet implemented with the embed webapp and not required in case of most of the partners
+const tokenRequestCallBack = () => {}
+
+GSU.createIframe({
+  targetElementId: 'iframe-target',
+  targetPage: 'discover',
+  partnerId: 'gsudemo',
+  navigationCallBack,
+  tokenRequestCallBack,
+  // This targetUrls object isn't necessary in production integrations, but it allows us to point to the dev environment.
+  targetUrls: {
+    discovery: 'https://embed-webapp.www.gsudevelopment.com/discovery/{partnerId}',
+    watch: 'https://embed-webapp.www.gsudevelopment.com/watch/{partnerId}/{classId}',
+  },
+})
```

Comment:
We should fix it in the package, and make it `discovery` if that's what we have in the URL.